scripts/research/lib/batch02_evidence_retention.py

```
@@ -0,0 +1,1036 @@
+        env.pop(key, None)
+    return env
+
+
+def run_git(cwd: Path, args: Sequence[str]) -> bytes:
+    """Run git with an argument array in an isolated working directory."""
+    _assert_safe_git_args(args)
+    if cwd.is_symlink():
+        raise PreOutcomeRetentionError("git cwd may not be a symlink")
+    cmd = ["git", "-C", str(cwd), *args]
+    try:
+        completed = subprocess.run(
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)